src/unix/stream.c

```diff
@@ -1602,6 +1602,32 @@ int uv_read_stop(uv_stream_t* stream) {
   return 0;
 }
 
+int uv_export(uv_stream_t* stream, uv_stream_info_t* info) {
+  int fd;
+
+  if (stream->type != UV_TCP)
+    return UV_EINVAL;
+
+  fd = uv__dup(stream->io_watcher.fd);
+
+  if (fd < 0)
+    return fd;
+
+  info->type = stream->type;
+  info->fd = fd;
+
+  return 0;
+}
+
+int uv_import(uv_stream_t* stream, uv_stream_info_t* info) {
+  if (info->type != UV_TCP)
```

Comment:
```suggestion
  if (stream->type != UV_TCP || info->type != UV_TCP)
```
That's what the win32 implementation does.

Design issue: I see the expectation is that the handle is pre-initialized but I think it makes more sense to let libuv initialize it (and change `uv_import()` to take a `uv_loop_t*` arg) because of this:
```c
uv_pipe_t one, two;
uv_stream_info_t info_one, info_two;

uv_pipe_init(uv_default_loop(), &one, /* ipc */ 0);
uv_pipe_init(uv_default_loop(), &two, /* ipc */ 1);

// swap handles
uv_export((uv_stream_t*) &one, info_one);
uv_export((uv_stream_t*) &two, info_two);
uv_import((uv_stream_t*) &two, info_one);  // no longer matches 'ipc'
uv_import((uv_stream_t*) &one, info_two);  // ditto
```
Similar reasoning applies to e.g. `uv_udp_init_ex()` (forgets the `flags` arguments.)